fl4health/losses/weight_drift_loss.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+import torch
+import torch.nn as nn
+
+
+class WeightDriftLoss(nn.Module):
+    def __init__(
+        self,
+        device: torch.device,
+    ) -> None:
+        super().__init__()
+        self.device = device
+
+    def _compute_weight_difference_inner_product(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        return torch.pow(torch.linalg.norm(x - y), 2.0)
+
+    def forward(self, target_model: nn.Module, constraint_tensors: List[torch.Tensor], weight: float) -> torch.Tensor:
```

Comment:
Note, the weight here has to be provided in the forward to be compatible with FedProx adaptive weight functionality. That is, setting the weight in the __init__ wouldn't work the way we wanted because FedProx (or other methods), might modify it.